### PR TITLE
REP-6337 Convert SchemaTests to use the ConfigurationTest parent

### PR DESCRIPTION
**Pull request by dmnjohns**

### CODE REVIEW

File: repose-aggregator/components/filters/add-header-filter/src/test/scala/org/openrepose/filters/addheader/config/AddHeaderSchemaTest.scala [hunk 1]
```diff
@@ -19,21 +19,20 @@
...
-import org.scalatest.{FunSpec, Matchers}
 import org.xml.sax.SAXParseException
 
 @RunWith(classOf[JUnitRunner])
-class AddHeaderSchemaTest extends FunSpec with Matchers {
-  val validator = ConfigValidator("/META-INF/schema/config/add-header.xsd")
+class AddHeaderSchemaTest extends ConfigurationTest {
+  override val schema: URL = getClass.getResource("/META-INF/schema/config/add-header.xsd")
+  override val exampleConfig: URL = getClass.getResource("/META-INF/schema/examples/add-header.cfg.xml")
+  override val jaxbContextPath: String = classOf[ObjectFactory].getPackage.getName
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/add-header-filter/src/test/scala/org/openrepose/filters/addheader/config/AddHeaderSchemaTest.scala]: Lol, you used that pointless ObjectFactory

Comment (dmnjohns) [reply repose-aggregator/components/filters/add-header-filter/src/test/scala/org/openrepose/filters/addheader/config/AddHeaderSchemaTest.scala]: I did, kind of. I really just needed the bound package name, but hey, if I can get that from a consistently generated class, I'll take that over a string that would also be a second source of truth. :)

File: repose-aggregator/components/filters/header-user-filter/src/test/scala/org/openrepose/filters/headeruser/config/HeaderUserSchemaTest.scala [hunk 2]
```diff
@@ -20,21 +20,20 @@
 
 package org.openrepose.filters.headeruser.config
```
Comment (wdschei) [inline repose-aggregator/components/filters/header-user-filter/src/test/scala/org/openrepose/filters/headeruser/config/HeaderUserSchemaTest.scala]: We should standardize on where we are putting this test. Just over half are in the `org.openrepose.filters.filtername` package (16) and others are a layer deeper in the `org.openrepose.filters.filtername.config` package (14).

File: repose-aggregator/components/filters/body-patcher-filter/src/test/scala/org/openrepose/filters/bodypatcher/BodyPatcherSchemaTest.scala [hunk 3]
```diff
@@ -19,18 +19,22 @@
...
 /**
   * Created by adrian on 4/29/16.
   */
 @RunWith(classOf[JUnitRunner])
-class SchemaAssertionTest extends FunSpec with Matchers {
-  val validator = ConfigValidator("/META-INF/schema/config/body-patcher.xsd")
+class BodyPatcherSchemaTest extends ConfigurationTest {
+  override val schema: URL = getClass.getResource("/META-INF/schema/config/body-patcher.xsd")
+  override val exampleConfig: URL = getClass.getResource("/META-INF/schema/examples/body-patcher.cfg.xml")
+  override val jaxbContextPath: String = classOf[ObjectFactory].getPackage.getName
```
[the remark this answers is not in the document]
Comment (dmnjohns) [reply repose-aggregator/components/filters/body-patcher-filter/src/test/scala/org/openrepose/filters/bodypatcher/BodyPatcherSchemaTest.scala]: True story.

File: repose-aggregator/components/filters/header-user-filter/src/test/scala/org/openrepose/filters/headeruser/config/HeaderUserSchemaTest.scala (hunk 2 above)
Comment (dmnjohns) [reply repose-aggregator/components/filters/header-user-filter/src/test/scala/org/openrepose/filters/headeruser/config/HeaderUserSchemaTest.scala]: That's fair. Sticking it in the `config` package means that I don't have to import the `ObjectFactory` in most if not all cases, so I would suggest putting them there.

I'll move them.

**Comment by dmnjohns** [comment]
Turns out, we also had some duplicate tests in there. I think they were the ones written in Java that we duplicated in an effort to test all schemas. Anyway, I've consolidated those now, and also stuck all of the config stuff into the config package. Yay consistency!

**Comment by wdschei** [comment]
I was just making sure we had all of the tests covered and I found a few missing and one that has a name inconsistent with the others.

The poorly named one is `ValkyrieAuthorizationFilterSchemaTest.scala`. This is the only Schema Test with `Filter` in it's name. Odd. Not show stopping, just odd.

I know that not all of the XSD's need a test specifically for only them, but I wanted to get the list out here so we could get a story going to get these missing tests added.

Here is the list of what is in the Core API:

* httpx/httpx-headers.xsd
* httpx/httpx-common.xsd
* httpx/httpx.xsd
* metrics/metrics.xsd

Here is the list from the filters:

* api-validator-filter (validator-configuration.xsd)
* cors-filter (cors-configuration.xsd)
* destination-router-filter (destination-router-configuration.xsd)
* herp-filter (highly-efficient-record-processor.xsd)
* merge-header-filter (merge-header.xsd)
* scripting-filter (scripting.xsd)
* translation-filter (translation-configuration.xsd)
     * (xproc/xml.xsd)
     * (xproc/xproc-papi.xsd)
     * (xproc/xproc.xsd)
* uri-normalization-filter (uri-normalization-configuration.xsd)
* url-extractor-to-header-filter (url-extractor-to-header.xsd)

**Comment by dmnjohns** [comment]
To follow up on that, I'll go ahead and rename the `ValkyrieAuthorizationFilterSchemaTest` test.

I'll also note that the `api-validator-filter` does have a test, the name was just typo'd. I'll fix that, too.

The rest is legit!

**Pull request merged**